src/Microsoft.DotNet.Watcher.Tools/DotNetWatcher.cs

```diff
@@ -1,176 +1,80 @@
-                        {
-                            _logger.LogInformation($"dotnet exit code: {dotnetExitCode}");
-                        }
-                        else
-                        {
-                            _logger.LogError($"dotnet exit code: {dotnetExitCode}");
-                        }
-
-                        _logger.LogInformation("Waiting for a file to change before restarting dotnet...");
-                        // Now wait for a file to change before restarting dotnet
-                        changedFile = await WaitForProjectFileToChangeAsync(projectFile, cancellationToken);
-                    }
-                    else
-                    {
-                        // This is a file watcher task
-                        changedFile = fileWatchingTask.Result;
+                        return;
                     }
 
-                    if (!string.IsNullOrEmpty(changedFile))
+                    if (finishedTask == processTask)
                     {
-                        _logger.LogInformation($"File changed: {changedFile}");
-                    }
-                }
-            }
-        }
-
-        private async Task<string> WaitForProjectFileToChangeAsync(string projectFile, CancellationToken cancellationToken)
-        {
-            using (var projectWatcher = CreateProjectWatcher(projectFile, watchProjectJsonOnly: false))
-            {
-                return await projectWatcher.WaitForChangeAsync(cancellationToken);
-            }
-        }
-
-        private Task<int> WaitForDotnetToExitAsync(string dotnetArguments, string workingDir, CancellationToken cancellationToken)
-        {
-            _logger.LogInformation($"Running dotnet with the following arguments: {dotnetArguments}");
-
-            var dotnetWatcher = _processWatcherFactory();
-            int dotnetProcessId = dotnetWatcher.Start("dotnet", dotnetArguments, workingDir);
-            _logger.LogInformation($"dotnet process id: {dotnetProcessId}");
-
-            return dotnetWatcher.WaitForExitAsync(cancellationToken);
-        }
-
-        private async Task WaitForValidProjectJsonAsync(string projectFile, CancellationToken cancellationToken)
-        {
-            while (true)
-            {
-                IProject project;
-                string errors;
-                if (_projectProvider.TryReadProject(projectFile, out project, out errors))
-                {
-                    return;
-                }
+                        _logger.LogInformation("Waiting for a file to change before restarting dotnet...");
 
-                _logger.LogError($"Error(s) reading project file '{projectFile}': ");
-                _logger.LogError(errors);
-                _logger.LogInformation("Fix the error to continue.");
+                        // Now wait for a file to change before restarting process
+                        await Task.WhenAll(fileSetTask).ConfigureAwait(false);
```

Comment:
There is just one task - can't you just do `await fileSetTask.ConfigureAwait(false)` ?